iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -535,57 +538,60 @@ string<Capacity>::insert(const uint64_t pos, const T& str, const uint64_t count)
     return true;
 }
 
+// AXIVION Next Construct AutosarC++19_03-A14.7.1:
 template <uint64_t Capacity>
-inline optional<string<Capacity>> string<Capacity>::substr(const uint64_t pos, const uint64_t count) const noexcept
+inline auto string<Capacity>::substr(const uint64_t pos, const uint64_t count) const noexcept
+    -> optional<string<Capacity>>
 {
     if (pos > m_rawstringSize)
     {
         return nullopt;
     }
 
-    uint64_t length = std::min(count, m_rawstringSize - pos);
+    uint64_t length{std::min(count, m_rawstringSize - pos)};
     string subString;
     std::memcpy(&(subString.m_rawstring[0]), &m_rawstring[pos], length);
     subString.m_rawstring[length] = '\0';
     subString.m_rawstringSize = length;
     return subString;
 }
 
+// AXIVION Next Construct AutosarC++19_03-A14.7.1:
```

Comment:
same here

Reply:
@FerdinandSpitzschnueffler right, it is suppressed globally